pipeline/summarize_results.py

```diff
@@ -31,6 +32,52 @@
 from nemo_skills.evaluation.metrics import MathEval
 from nemo_skills.utils import setup_logging
 
+LOG = logging.getLogger(__name__)
+
+
+def process_batch_results(prediction_jsonl_files):
+    for batch_request_file in prediction_jsonl_files:
+        jsonl_file = batch_request_file.with_name(batch_request_file.name.replace('.jsonl-batch-request-id', '.jsonl'))
+
+        if batch_request_file.exists():
+            try:
+                with open(batch_request_file, 'rt', encoding='utf-8') as fin:
+                    request_id = json.load(fin)['request_id']
+
+                from nemo_skills.inference.server.model import get_model
+
+                llm = get_model(server_type='openai', model='gpt-4-1106-preview')
+                metadata, outputs = llm.get_batch_results(request_id)
+
+                if outputs is None:
+                    LOG.warning(f"Judgements are not ready yet for {jsonl_file}! Current status: {metadata}")
+                    continue
+
+                # Read existing data from the jsonl file
+                with open(jsonl_file, 'rt', encoding='utf-8') as fin:
+                    data = [json.loads(line) for line in fin]
+
+                # Update data with judgements
+                for data_point, output in zip(data, outputs):
+                    data_point['judgement'] = output['generation']
```

Comment:
Let's use the "generation_key" here that you would add into the batch-request-id file